src/test/scala/chiselTests/BundleLiterals.scala

```diff
@@ -0,0 +1,66 @@
+// See LICENSE for license details.
+
+package chiselTests
+
+import chisel3._
+import org.scalatest._
+
+class BundleWithLit(lit: Option[Int]=None) extends Bundle {
+  val internal = lit.map(_.U).getOrElse(UInt(4.W))
+  override def cloneType = new BundleWithLit(lit).asInstanceOf[this.type]
+}
+
+object BundleWithLit {
+  def apply(x: Int) = {
+    val ret = new BundleWithLit(Some(x))
+    Aggregate.LitBind(ret)
+    ret
+  }
+  def apply() = {
+    new BundleWithLit()
+  }
+}
+
+class BundleWithBundleWithLit(lit: Option[Int]=None) extends Bundle {
+  val internal = lit.map(BundleWithLit(_)).getOrElse(BundleWithLit())
+  override def cloneType = new BundleWithBundleWithLit(lit).asInstanceOf[this.type]
+}
+
+object BundleWithBundleWithLit {
+  def apply(x: Int) = {
+    val ret = new BundleWithBundleWithLit(Some(x))
+    Aggregate.LitBind(ret)
+    ret
+  }
+  def apply() = {
+    new BundleWithBundleWithLit()
+  }
+}
+
+class BundleWithLitModule extends Module {
+  val io = IO(new Bundle {
+    val out = Output(BundleWithLit())
+  })
+  io.out := BundleWithLit(3)
```

Comment:
Is this the intended bundle literal syntax? Also, what about bundles with multiple members?

Reply:
I don't know if I'd say *the* literal syntax, but it is the kind of syntax we use with DspReal().
You're companion object could take multiple parameters